/* coding:utf-8
 * Copyright (C) dirlt
 */

package com.dirlt.java.FastHbaseRest;

option java_outer_classname = "MessageProtos1";

// --------------------
message ReadRequest {  
  required string table_name = 1;
  required string row_key = 2;
  required string column_family = 3;
  // if you just want to query column faimly
  // then don't set qualifiers.
  // but if you query column family, we don't do cache.
  repeated string qualifiers = 4;
  optional int32 timeout = 5;
  optional int32 retry = 6;
}

message ReadResponse {
  message KeyValue {
    // order as request.!!!
    // if empty, return empty bytes.
    required string qualifier = 1;
    required bytes content = 2;
  }
  repeated KeyValue kvs = 4;
  required bool error = 5;
  optional string message = 6;
}

// --------------------
message MultiReadRequest {
  repeated ReadRequest requests = 1;
  optional int32 timeout = 2;
  optional int32 retry = 3;
}

message MultiReadResponse {
  repeated ReadResponse responses = 1;
}

// --------------------
message WriteRequest {
  required string table_name = 1;
  required string row_key = 2;
  required string column_family = 3;    
  message KeyValue {
    required string qualifier = 1;
    required bytes content = 2;
  }
  repeated KeyValue kvs = 4;
  optional int32 timeout = 5;
  optional int32 retry = 6;
}

// empty message.
message WriteResponse {
   required bool error = 1;
   optional string message = 2;
}

// --------------------
message MultiWriteRequest {
  repeated WriteRequest requests = 1;
  optional int32 timeout = 2;
  optional int32 retry = 3;
}

message MultiWriteResponse {
  repeated WriteResponse responses = 1;
}